dfsclient/src/main/java/org/hpccsystems/dfs/client/AvroRecordTranslator.java

```diff
@@ -0,0 +1,1042 @@
+                {
+                    Number numVal = (Number) field;
+                    return Integer.valueOf(numVal.intValue());
+                }
+                else
+                {
+                    throw new Exception("AvroFieldTranslator toAvro: Integer: Unexpected value type: " + field.getClass());
+                }
+            }
+            case LONG:
+            {
+                if (fd.getFieldType() != FieldType.INTEGER)
+                {
+                    throw new Exception("AvroFieldTranslator toAvro: incompatible avro schema type: " 
+                                        + schema.getType() + "  to hpcc field type: " + fd.getFieldType());
+                }
+
+                if (field instanceof Number)
+                {
+                    Number numVal = (Number) field;
+                    return Long.valueOf(numVal.longValue());
+                }
+                else
+                {
+                    throw new Exception("AvroFieldTranslator toAvro: Long: Unexpected value type: " + field.getClass());
+                }
+            }
+            case MAP:
+            {
+                if (fd.getFieldType() != FieldType.DATASET)
+                {
+                    throw new Exception("AvroFieldTranslator toAvro: incompatible avro schema type: " 
+                                        + schema.getType() + "  to hpcc field type: " + fd.getFieldType());
+                }
+
+                if (field instanceof List)
+                {
+                    HashMap<String,Object> map = new HashMap<String,Object>();
+                    List<HPCCRecord> recordList = (List<HPCCRecord>) field;
+                    for (int i = 0; i < recordList.size(); i++)
+                    {
+                        HPCCRecord rec = recordList.get(i);
+                        if (rec.getNumFields() != 2)
+                        {
+                            throw new Exception("AvroFieldTranslator toAvro: Map: Unexpected num fields in map record: " + rec.getNumFields());
+                        }
+
+                        String key = (String) rec.getField(0);
+                        Object val = toAvro(schema.getValueType(), fd.getDef(0).getDef(1),rec.getField(1));
+                        map.put(key,val);
+                    }
+
+                    return map;
+                }
+                else
+                {
+                    throw new Exception("AvroFieldTranslator toAvro: Map: Unexpected value type: " + field.getClass());
+                }
+            }
+            case NULL:
+            {
+                if (fd.getFieldType() != FieldType.BOOLEAN)
+                {
+                    throw new Exception("AvroFieldTranslator toAvro: incompatible avro schema type: " 
+                                        + schema.getType() + "  to hpcc field type: " + fd.getFieldType());
+                }
+
+                if (field instanceof Boolean)
```

Comment:
this mapping doesn't make 100% sense to me...
This case handles a NULL Avro type? and we map to ecl boolean?